h2o-core/src/main/java/water/H2O.java

```diff
@@ -1725,9 +1723,30 @@ public static void gc() {
     }
   }
 
+  /**
+   * Check if the Java version is not supported
+   * @return true if not supported
+   */
+  public static boolean checkUnsupportedJava() {
+    String version = System.getProperty("java.version");
+    if (version != null && !(version.startsWith("1.6") || version.startsWith("1.7") || version.startsWith("1.8"))) {
```

Comment:
This is ok now with the null check but for the future you can also use `org.apache.commons.lang.SystemUtils.IS_JAVA_1_6` etc.